backend/accounts/views.py

```diff
@@ -527,3 +528,19 @@ def post(self, request, format=None):
                     )
                     user.user_permissions.add(permission)
         return Response({"detail": "success"})
+
+
+class PrivacySettingView(
+    generics.GenericAPIView, mixins.ListModelMixin, mixins.UpdateModelMixin
+):
+    serializer_class = PrivacySettingSerializer
+    permission_classes = [IsAuthenticated]
+
+    def get_queryset(self):
+        return self.request.user.privacy_setting.all()
```

Comment:
You might want to use `select_related` and fetch privacy resources here, to avoid an N+1 while serializing.

Reply:
Thanks for the catch!